### ISSUE TITLE
Refactor: Project Creation

### ISSUE THREAD

**Issue by AmineOuerfellii** [activetigger/activetigger#979]
Refactor: Improve Project Creation Dataset Splitting Strategy

Related to #934 

### Explanation 

The core idea was to give the user full freedom over how they split their dataset. To do that properly, i had to think through every possible situation a user could face:

- A user with :  (only train data) or train+(val/+test)

once i mapped out all these cases, a natural two-axis structure emerged as follow:

1st Axis : Train Selection(`build_trainset`):
- `sequential` :first N rows, deterministic
- `force_label`:prioritize already labeled rows
- `stratify`:stratified split by column
- `random` :default fallback

2ndAxis  :Holdout Selection(`build_test_val_set`):
- `sequential` :supports custom `start_index_val` + `start_index_test` with  overlap protection (train/valid/test overlap checks) as this is  needed because when splits are deterministic and user sets custom indices, overlaps are possible and silent, so we raise explicitly , for the start_index (vat/test ) the idea is the user could bring a dataset with a certain index in head and a certain size of test/validation sample , of course in the case of deterministic split 
- `stratify` — stratified holdout draw
- `random` — default holdout draw

sequential train : build TRAIN first  -> holdout from remaining
non-sequential: build HOLDOUT first -> train from remaining

### Changes

- Refactored `create_project` task into `build_trainset` + `build_test_val_set` methods
- Updated `ProjectCreationForm` with new split options
- Improved `ProjectCreationForm` UI/UX for better readability and aesthetics
- Updated `ProjectParameters` display
- Updated styles for new form layout
- User can now **automatically select the suggested file** (`gwsd_train_test_dataset`) , no need to download it manually
- **Label column suggestion** :automatically detects potential label columns
----
# Demonstration :
Branch based on commit bba30494  "add contribution.md"
[Branch](https://github.com/AmineOuerfellii/activetigger/tree/refactor/ProjectCreation)

<img width="500" height="600" alt="Image" src="https://github.com/user-attachments/assets/a68fe91a-4ff1-469f-991b-73e44069f829" />[Project Parameters Display]

![Image](https://github.com/user-attachments/assets/b95547e6-4cb1-47b3-ae10-ad67272b80a7)

**Comment by emilienschultz** [comment]
It is an interesting discussion you are opening. Thanks.

Regarding the refactor, we spent some time to manage corner case, so I would like to be sure that we don't introduce uncontrolled situations. That's why I am being a bit cautious.

The main point is that the valid/test dataset should be handle before the train set with random (stratified if selected), except when the draw is no more random (what you called sequential). In this case, we would like to deactivate valid and test, since it is no more possible to assure their representativity. (btw, I like the idea of calling it sequential). 

Regarding the new design, I like the general idea and I think it is a good idea. I am not sure for the selectors for the train/valid/test because it introduce the idea we have a proportion of the total (which make no sense for large datasets of million of elements). What do you think @jboelaert @AXLMRIN 

How do you manage the label suggestion ? Again, it is a nice idea, so I would say rather yes.

If you want to open a PR on this specific feature (refactor create_project task and frontent) I can conduct some tests (and wait if Julien or Axel has any additional thoughts on it)

**Comment by AXLMRIN** [comment]
Hello!
Thank you for all this work, I agree that this page needs some love. I will give it some thoughts on Tuesday and get back to you! 

Have you open a PR so I can test it? 

Anyhow, thank you for your commitment!
AM

**Comment by AmineOuerfellii** [comment]
Hello everyone,  i'm so glad that my contribution is having such a positive feedback ,the issue raised by @jboelaert def worth it .As i mentionned before project creation is now built around two independant axes which results in : 4 train options * 3 holdout option  meaning 12 possible combos in total .

Regarding the excellent point raised by @emilienschultz and i quote:
>The main point is that the valid/test dataset should be handle before the train set with random (stratified if selected), except when the draw is no more random (what you called sequential). In this case, we would like to deactivate valid and test, since it is no more possible to assure their representativity. (btw, I like the idea of calling it sequential)

When a user selects sequential train + random/stratified holdout, the holdout is drawn from the remaining rows after excluding the first n_train rows. As a result, the evaluation set is representative of the data excluding the initial train block, not necessarily the full corpus,
since choosing "sequential" is a deliberate decision that respects the user's own data ordering, ensuring full representativity becomes his responsibility too because that implies that  they have a specific reason for that ordering and they know what they are doing with their data so rather than deactivating valid/test in sequential mode, i kept them active 

This design also supports other cases with pre-prepared datasets:

- The user can place their training rows at the top , set train_selection = "sequential" with a chosen n_train, and then define starting indices for validation and test (start_index_val and start_index_test) and the system will take exactly the slices they specify for valid and test and this calls for the overlap protection as there is an explicit overlap check between train, validation, and test sets. If any overlap is detected, the project creation fails with a clear error

- Automatic fallback (simple case): if no custom start indices are provided, validation starts right after the train set (start_val = n_train), and test starts after validation. This is convenient when the user has simply ordered their file as: [Train rows] -> [Validation rows] -> [Test rows] 

Or simply only train set (no evaluation):
The user can simply set n_valid = 0 and n_test = 0. In this case, no holdout is created, and the entire focus is on the training set even the holdout selection div in frontend disappears and reappears once the user changes/updates the val / test size , and more ...

For the automatic label column detection, i decided to handle it in the frontend rather than in the backend  which was where i started implementing it at first and when i begun like looking at it from another POV i found it to be heavy and over engineered for a simple task like that 

any way i'll open a PR right now for everyone to check it :

i 'm always open to discuss this further and would be happy to adjust based on your feedback

Regards

**Comment by AmineOuerfellii** [comment]
Hello , 

i've opened a PR right now for everyone to check it, i really appreciate the feedback, and the recommendations , notes were taken into consideration , and i hope it leads to a more functional approach overall

Regards
